src/coreclr/src/vm/jithelpers.cpp

```diff
@@ -5133,6 +5133,23 @@ Thread * __stdcall JIT_InitPInvokeFrame(InlinedCallFrame *pFrame, PTR_VOID StubS
 EXTERN_C void JIT_PInvokeBegin(InlinedCallFrame* pFrame);
 EXTERN_C void JIT_PInvokeEnd(InlinedCallFrame* pFrame);
 
+EXTERN_C void JIT_ReversePInvokeEnter()
+{
+    Thread* thread = GetThreadNULLOk();
+    if (thread == NULL)
+        thread = CreateThreadBlockThrow();
+
```

Comment:
The existing slow implementation has sanity check for bad ReversePInvoke use (look for `ReversePInvokeBadTransition`). It costs like two extra instructions, but I think is very well worth it. It would be nice to preserve it for this fast path.